content/post/39-go-libp2p-6.0.0.md

```diff
@@ -0,0 +1,310 @@
+---
+date: 2018-06-15
+url: 39-go-libp2p-6-0-0
+title: go-libp2p 6.0.0 released
+author: Steven Allen
+---
+
+# 🔦 go-libp2p 6.0.0 release notes
+
+We're pleased to announce go-libp2p 6.0.0. This release includes a massive
+refactor of go-libp2p that paves the way for new transports such as QUIC.
+Unfortunately, as it is broad sweeping, there are some breaking changes,
+*especially* for maintainers of custom transports.
+
+Below, we cover the changes you'll likely care about. For convenience, we've
+broken this into a section for users and transport authors/maintainers. However,
+transport maintainers should really read both sections.
+
+## 💁 For Users
+
+Libp2p users should be aware of a few major changes.
+
+* Guarantees and performance concerning connect/disconnect notification
+  processing have improved.
+* Handling of half-closed streams has changed (READ THIS SECTION).
+* Some constructors and method signatures have changed slightly.
+
+### Dialing And Source Addresses
+
+We've improved the logic that selects the source address when dialing. In the
+past, you may have run into an issue where you couldn't dial non-local nodes
+when listening on 127.0.0.1 as opposed to 0.0.0.0. This happened because
+go-libp2p would randomly pick the source address from the set of addresses on
+which the node was listening. We did this to ensure that the other end could
+dial us back at the same address. Unfortunately, one can't use 127.0.0.1 as a
+source address when dialing any non-local address so outbound dials failed.
+
+go-libp2p now tries to be smarter about this and avoids picking source addresses
+that have no route to the destination address.
+
+### Bandwidth Metrics
+
+To start out on an unhappy note, bandwidth metrics are now less accurate. In the
+past, transports returned "minimal" connections (e.g., a TCP connection) so we
+could wrap these transport connections in "metrics" connections that counted
+every byte sent and received.
+
+Unfortunately, now that we've moved encryption and multiplexing down into the
+transport layer, the connection we're wrapping has significantly more
+under-the-covers overhead.
+
+However, we do hope to improve this and get even *better* bandwidth metrics than
+we did before. See [libp2p/go-libp2p-transport#31][] for details.
+
+[libp2p/go-libp2p-transport#31]: https://github.com/libp2p/go-libp2p-transport/issues/31
+
+### Notifications
+
+This release brings performance improvements and easy to reason about ordering
+guarantees libp2p connect/disconnect events:
```

Comment:
wording here sounds weird.

Reply:
weird = missing a word.

How about "This release brings performance improvements and simple ordering guarantees for connect/disconnect events"?

Reply:
I've taken a stab at rewording this for easier reading.

Reply:
New commit - adopted @Stebalien's wording in latest commit.